crates/consensus/rpc/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+/// TODO: Crate description
+
+use alloy_consensus::TxEnvelope;
+use alloy_eips::eip2718::Encodable2718;
+use reth_node_api::EngineTypes;
+use reth_node_core::{
+    primitives::{B256},
+    rpc::types::{BlockTransactions, ExecutionPayloadV2, ExecutionPayloadV3, RichBlock},
+};
+use reth_rpc_builder::auth::AuthServerHandle;
+use reth_rpc_types::ExecutionPayloadV1;
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::time::sleep;
+
+/// Fake consensus client that sends FCUs and new payloads using recent blocks from an external
+/// provider, starting with Etherscan.
+/// TODO: Naming - maybe "ExternalConsensusClient"? Or maybe simpler and just
+///   "EtherscanConsensusClient"?
+#[derive(Debug)]
+pub struct RpcConsensusClient {
+    /// HTTP client to fetch blocks
+    http_client: reqwest::Client,
+    /// Handle to execution client
+    auth_server: AuthServerHandle,
+    /// Etherscan API key
+    etherscan_api_key: String,
+    /// Etherscan base API URL
+    etherscan_base_api_url: String,
+}
+
+impl RpcConsensusClient {
+    /// Create a new fake consensus client that should sent FCUs and new payloads to `auth_server`.
+    pub fn new(
+        auth_server: AuthServerHandle,
+        etherscan_api_key: String,
+        etherscan_base_api_url: String,
+    ) -> Self {
+        Self {
+            http_client: reqwest::Client::new(),
+            auth_server,
+            etherscan_api_key,
+            etherscan_base_api_url,
+        }
+    }
+
+    /// Spawn the client to start sending FCUs and new payloads by periodically fetching recent blocks.
+    pub async fn spawn<T: EngineTypes>(&self) {
+        // TODO: Add logs
+        // TODO: Generalize over block-fetching code to support different sources
+
+        let execution_client = self.auth_server.http_client();
+        let mut last_block_number: Option<u64> = None;
+
+        loop {
+            let block: EtherscanBlockResponse = self
+                .http_client
+                .get(&self.etherscan_base_api_url)
+                .query(&[
+                    ("module", "proxy"),
+                    ("action", "eth_getBlockByNumber"),
+                    ("tag", "latest"),
+                    ("boolean", "true"),
+                    ("apikey", &self.etherscan_api_key),
+                ])
+                .send()
+                .await
+                .unwrap()
+                .json()
+                .await
```

Comment:
I'd like to make this configurable, for example I think it could be useful to provide another node's rpc endpoint and use eth_subscribe + eth_getBlockByHash to fetch the block

so perhaps a stream type helper that yields full blocks

this is perhaps a better solution that just etherscan,
see for example:

https://github.com/alloy-rs/examples/blob/51934948a20c6edadd1ca5ab04726962ee153d38/examples/providers/examples/builtin.rs#L16